io/snmp/snmp.js

```diff
@@ -52,14 +52,72 @@ module.exports = function(RED) {
             }
         });
     }
-    RED.nodes.registerType("snmp",SnmpNode);
+    RED.nodes.registerType("snmp", SnmpNode);
+
+
+    function SnmpSNode(n) {
+        RED.nodes.createNode(this, n);
+        this.community = n.community;
+        this.host = n.host;
+        this.version = (n.version === "2c") ? snmp.Version2c : snmp.Version1;
+        var node = this;
+        var objectTypes = {
+            "Boolean": 1,
+            "Integer": 2,
+            "OctetString": 4,
+            "Null": 5,
+            "OID": 6,
+            "IpAddress": 7,
+            "Counter": 8,
+            "Gauge": 66,
+            "TimeTicks": 67,
+            "Opaque": 68,
+            "Counter64": 70,
+            "NoSuchObject": 128,
+            "NoSuchInstance": 129,
+            "EndOfMibView": 130
+        };
+
+        this.on("input", function (msg) {
+            var host = msg.host || node.host;
+            var community = msg.community || node.community;
+            var varbinds = msg.varbinds;
+
+            if (varbinds) {
+                for (var i = 0; i < varbinds.length; i++) {
```

Comment:
ObjectType also has the keys/value pairs switched. You don't need an `eval()`. You can use the line as I wrote it in the above comment.